### ISSUE TITLE
Top & Bottom Parts of the image is "lost" when zooming in

### ISSUE THREAD

**Issue by wooldridgetm** [renancaraujo/photo_view#11]
I discovered that when the height of the PhotoView isn't set to full-screen, then after zooming in, the top & bottom parts of the image aren't visible to the user.

I think it's because in **_PhotoViewImageWrapperState**, **Fx _clampPosition(Offset)**, we're using the screen's width & height instead of the UI Component's width & height.

`final screenHeight = MediaQuery.of(context).size.height` _needs to be_ `final screenHeight = context.size.height` (and likewise for width).

Hopefully that all makes sense.
Great job on this, btw.  Works great.

**Comment by renancaraujo** [comment]
You are right. I think we should include an example when photo view isn't used as a full-screen widget.

**Comment by wooldridgetm** [comment]
I can provide you a link to a repo with an example if you'd like.

**Comment by renancaraujo** [comment]
I will see what we can do to avoid that. Further PR is coming.

**Comment by sgon00** [comment]
This problem does not only happen when zooming in, `PhotoViewScaleBoundary.contained` will also make the top and bottom parts lost. Check the following screenshots:

![contained](https://user-images.githubusercontent.com/1555508/43948290-06f7f57c-9cbd-11e8-89b4-25e63edc55d7.png)
![entire_image](https://user-images.githubusercontent.com/1555508/43948292-08873e20-9cbd-11e8-8af4-853b7bab020b.png)

**Comment by renancaraujo** [comment]
Could you upload the code of this example perhaps?
It would help me to figure out what is causing this weird behavior.

**Comment by sgon00** [comment]
@renancaraujo sure, the code is very simple. 

### main.dart

```dart
import 'package:flutter/material.dart';
import 'package:photo_view/photo_view.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Bug Demo',
      theme: ThemeData(
        primarySwatch: Colors.teal,
      ),
      home: Scaffold(
        appBar: AppBar(
          title: Text(''),
        ),
        body: Container(
          child: PhotoView(
            minScale: PhotoViewScaleBoundary.contained,
            maxScale: 3.0,
            backgroundColor: Color.fromRGBO(255, 255, 255, 1.0),
            loadingChild: Center(child: CircularProgressIndicator()),
            imageProvider: AssetImage('assets/100x500.png'),
          )
        ),
      ),
    );
  }
}
```

### pubspec.yaml

```
dependencies:
  photo_view: ^0.0.4

  assets:
   - assets/100x500.png
```

### assets/100x500.png

![100x500](https://user-images.githubusercontent.com/1555508/43962143-831f068c-9ce9-11e8-9d81-51527b8562f9.png)
